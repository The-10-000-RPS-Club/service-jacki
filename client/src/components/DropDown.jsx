/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function DropDown({ sort, setSort }) {
  const [dropDownContent] = useState(['Newest questions', 'Newest answers', 'Most answered', 'Answers needed', 'Most helpful answers']);

  return (
    <div>
      <NavbarDropdown>
        <p>
          Sort by: {sort} &#x25BE;
        </p>
        <NavbarDropdownContent>
          <SingleDropdownOption onClick={() => { setSort('Newest questions'); }}><div>{dropDownContent[0]}</div></SingleDropdownOption>
          <SingleDropdownOption onClick={() => { setSort('Newest answers'); }}><div>{dropDownContent[1]}</div></SingleDropdownOption>
          <SingleDropdownOption onClick={() => { setSort('Most answered'); }}><div>{dropDownContent[2]}</div></SingleDropdownOption>
          <SingleDropdownOption onClick={() => { setSort('Answers needed'); }}><div>{dropDownContent[3]}</div></SingleDropdownOption>
          <SingleDropdownOption onClick={() => { setSort('Most helpful answers'); }}><div>{dropDownContent[4]}</div></SingleDropdownOption>
        </NavbarDropdownContent>
      </NavbarDropdown>
    </div>
  );
}

DropDown.propTypes = {
  sort: PropTypes.string.isRequired,
  setSort: PropTypes.func.isRequired,
};

const NavbarDropdownContent = styled.div`
display: none;
position: absolute;
background-color: white;
border: 1px solid black;
min-width: 160px;
box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0, 2);
z-index: 9;
cursor: pointer;
font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
`;

const NavbarDropdown = styled.div`
position: relative;
display: inline-block;
float: right;
padding-bottom: 5px;
cursor: pointer;
z-index: 9;
&:hover ${NavbarDropdownContent} {
display: inline-block;
}
font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
`;

const SingleDropdownOption = styled.div`
border: 10px solid white;
&:hover {
background: #2B455C;;
color: white;
cursor: pointer;
border: 10px solid darkslategrey;
}
`;

export default DropDown;
