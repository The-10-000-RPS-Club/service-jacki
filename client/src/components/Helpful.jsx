/* eslint-disable no-underscore-dangle */
import React, { useState, useEffect } from 'react';

import styled from 'styled-components';

const Helpful = ({ question, incrementHelpfulCount }) => {
  const [quesId] = useState(question.question_id);
  const [ansId] = useState(question.answers[0]._id);
  const [yes, setYes] = useState(question.answers[0].helpful.yes);
  const [no, setNo] = useState(question.answers[0].helpful.no);

  return (
    <p>
      Helpful? <HelpfulButton className='yes' onClick={() => {
				incrementHelpfulCount(quesId, ansId, 'yes');
				setYes(yes + 1);
				}}>yes: {yes}</HelpfulButton> <HelpfulButton className='no' onClick={() => {
					incrementHelpfulCount(quesId, ansId, 'no');
					setNo(no + 1);
					}}>no: {no}</HelpfulButton> <HelpfulButton>Report as inappropriate</HelpfulButton>
    </p>
  );
};

const HelpfulButton = styled.button`
  background-color: white;
  border: 1px solid grey;
  border-radius: 2px;
  &:hover {
    box-shadow: inset 0 0 3px #000000;
  }
  position: relative;
`;

export default Helpful;
