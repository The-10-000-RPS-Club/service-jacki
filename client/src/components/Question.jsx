import React from 'react';

import moment from 'moment';

import styled from 'styled-components';

const Wrapper = styled.section`
	margin: auto;
	width: 50%;
	border-top: 1px solid grey;
	padding: 10px;
	position: relative;
	font-family: Stuart, Georgia, serif;
	padding-bottom: 10 px;
	margin-bottom: 10px;
`;

const HelpfulButton = styled.button`
	background-color: white;
	border: 1px solid grey;
	border-radius: 2px;
	&:hover {
    box-shadow: inset 0 0 3px #000000;
  }
`;

const AnswerNumber = styled.section`
	float: right;
	text-align: center;
	border-radius: 2px;
`;

const AnswerButton = styled.button`
	border: 1px solid grey;
	padding: 6px;
	background-color: white;
	margin: 10px;
	border-radius: 2px;
	margin-bottom: 40px;
`;

const QuestionTime = styled.button`
	font-family: Stuart, Georgia, serif;
	border: none;
`;

const User = styled.div`
	font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
	font-size: 15px;
	font-weight: 700;
`;

const QuestionBody = styled.div`
font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
font-size: 18px;
font-weight: 700;
margin-bottom: 40px;
`;

const AnswerBody = styled.div`
font-family: Stuart, Georgia, serif;
padding-bottom: 15px;
`;

const Question = (props) => (
	<Wrapper>
	<div>
		<AnswerNumber><p>{props.question.answers.length}</p><p>answers</p></AnswerNumber>
		<div><User>{props.question.user} &#183; <QuestionTime>{moment(props.question.created_at).startOf('hour').fromNow()}</QuestionTime></User></div>
		<QuestionBody><p href='#'>{props.question.question_body}</p></QuestionBody>
		<AnswerButton>Answer the question</AnswerButton>
	<div><User>{props.question.answers[0].user} &#183; <QuestionTime>{moment(props.question.answers[0].created_at).startOf('hour').fromNow()}</QuestionTime></User></div>
	<AnswerBody><p>{props.question.answers[0].body}</p></AnswerBody>
	<p>Helpful? <HelpfulButton className='yes' onClick={() => console.log(event.target.className)}>yes: {props.question.answers[0].helpful.yes}</HelpfulButton> <HelpfulButton className='no'>no: {props.question.answers[0].helpful.no}</HelpfulButton> <HelpfulButton>Report as inappropriate</HelpfulButton> </p>
	</div>
	</Wrapper>
);

export default Question;
