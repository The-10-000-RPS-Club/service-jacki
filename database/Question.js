const mongoose = require('mongoose');
const db = require('./index.js');

mongoose.Promise = global.Promise;

const questionSchema = new mongoose.Schema({
  question_id: Number,
  product_id: Number,
  user: String,
  question_body: String,
  answers: [
    {
      answer_id: Number,
      body: String,
      user: String,
      helpful: {
        yes: Number,
        no: Number,
      },
    },
  ],
},
{
  timestamps: true,
});

const Question = mongoose.model('Question', questionSchema);

module.exports = Question;
