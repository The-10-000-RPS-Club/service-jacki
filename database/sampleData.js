const sampleData = [
    {
question_id: 058, 
product_id: 017,
user: 'BobbyB',
question_body: 'Does this product fit both men and women?',
answer: [
    {
    id: 023,
    body: 'Yes it does.',
    user: 'REI Helper Sally',
    helpful: {
        yes: 0,
        no: 1,
    },
    },
],
}, 
{
    question_id: 009,
    product_id: 043,
    user: 'Rachel098',
    question_body: 'How durable is this?',
    answer: [
        {
        id: 023,
        body: 'It can stand up to all your favortie activities.',
        user: 'REI Helper Jim',
        helpful: {
            yes: 2,
            no: 0,
        },
        }, 
        {
            id: 079,
            body: 'It tore after one week.',
            user: 'CaliBro43',
            helpful: {
                yes: 1,
                no: 3,
            },
            },
    ],
    },
]