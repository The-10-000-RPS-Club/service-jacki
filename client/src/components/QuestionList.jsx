import React from 'react';
import styled from 'styled-components';
import Question from './Question.jsx';

const Wrapper = styled.section`
margin: auto;
width: 50%;
position: relative;
font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
font-size: 14px;
`;

const QuestionList = (props) => (
  <div>
	<Wrapper><p>{props.questions.length} Questions</p></Wrapper>
  {props.questions.map((question, i) => (
    <Question question={question} key={i} />
    ))}
  </div>
);

export default QuestionList;
