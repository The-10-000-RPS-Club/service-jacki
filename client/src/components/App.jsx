import React, { useState, useEffect } from 'react';

import axios from 'axios';

import styled from 'styled-components';

import QuestionList from './QuestionList.jsx';

import DropDown from './DropDown.jsx';

import LoadMore from './LoadMore.jsx';



function App() {

	const [questions, setQuestions] = useState([]);
	const [numQuestions, setNumQuestions] = useState(5);
	// const [sort, setSort] = useState(null);

	const getQuestions = () => {
		axios.get('/api/products/questions')
			.then((data) => setQuestions(questions.concat(data.data)))
			.catch((err) => (err));
	};
	
	const incrementHelpfulCount = (quesId, ansId, option) => {
		axios.patch(`/api/products/questions/${quesId}/${ansId}/${option}`)
			.then((data) => console.log(`${option} count increased`))
			.catch((err) => (err));
	};

	useEffect(() => {
    getQuestions();
	}, [numQuestions]);
	

console.log(numQuestions, questions.length);
  return (
    <div>
      <div>
        <Wrapper>
          <div>
            <QuestionButton>Ask a question</QuestionButton>
            <Title>
							<div>Questions & Answers</div>
							</Title>
           <DropDown />
          </div>
        </Wrapper>
        <div>
          <QuestionList questions={questions} incrementHelpfulCount={incrementHelpfulCount}/>
        </div>
      </div>
      <div>
      <LoadMore numQuestions={numQuestions} setNumQuestions={setNumQuestions} getQuestions={getQuestions}/>
			</div>
    </div>
  );
}

const Wrapper = styled.section`

margin: auto;
width: 55%;
position: relative;
padding: 10px;
font-family: Stuart, Georgia, serif;
`;

const Title = styled.div`
font-family: Stuart, Georgia, serif;
font-size: 22px;
padding 10px;
position: relative;
padding-bottom: 30px;
padding-left: 0px;
`;

const QuestionButton = styled.button`
background-color: #006699;
color: white;
float: right;
border: none;
padding: 10px;
border-radius: 2px;
font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
font-size: 16px;
`;

export default App;
