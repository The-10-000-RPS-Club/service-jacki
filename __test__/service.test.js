import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from '../client/src/components/App.jsx';
import QuestionList from '../client/src/components/QuestionList.jsx';
import Question from '../client/src/components/Question.jsx';
import Helpful from '../client/src/components/Helpful.jsx';
import DropDown from '../client/src/components/DropDown.jsx';
import sampleData from '../database/sampleData.js';

Enzyme.configure({ adapter: new Adapter() });

describe('App component tests', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<App />).contains(<div id="list-container">Bar</div>)).toBe(false);
  });
});

describe('A suite example using Snapshot',  () => {
  it('renders App component correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('renders Dropdown component correctly', () => {
	const tree = renderer
		.create(<DropDown />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders QuestionList component correctly', () => {
	const tree = renderer
		.create(<QuestionList questions={sampleData} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
