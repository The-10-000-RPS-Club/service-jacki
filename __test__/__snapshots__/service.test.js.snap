// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite example using Snapshot renders App component correctly 1`] = `
<div>
  <div>
    <section
      className="sc-gXfWUo fbgvgK"
    >
      <button
        className="sc-ciSmjq fGvxWW"
        onClick={[Function]}
      >
        Ask a question
      </button>
      <div>
        <div
          className="sc-cBoprd ebejrs"
        >
          <div>
            Questions & Answers
          </div>
        </div>
        <div>
          <div
            className="sc-iwaifL gUOPVN"
          >
            <p>
              Sort by: 
              -select-
               ▾
            </p>
            <div
              className="sc-bCwgka kfElgi"
            >
              <div
                className="sc-cxNIbT ibZciV"
                onClick={[Function]}
              >
                <div>
                  Newest questions
                </div>
              </div>
              <div
                className="sc-cxNIbT ibZciV"
                onClick={[Function]}
              >
                <div>
                  Newest answers
                </div>
              </div>
              <div
                className="sc-cxNIbT ibZciV"
                onClick={[Function]}
              >
                <div>
                  Most answered
                </div>
              </div>
              <div
                className="sc-cxNIbT ibZciV"
                onClick={[Function]}
              >
                <div>
                  Answers needed
                </div>
              </div>
              <div
                className="sc-cxNIbT ibZciV"
                onClick={[Function]}
              >
                <div>
                  Most helpful answers
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <div>
      <div>
        <section
          className="sc-fKgIGh keGHJm"
        >
          <p>
            1-
            0
             of 
            0
             Questions
          </p>
        </section>
      </div>
    </div>
  </div>
  <div>
    <button
      className="sc-lmgQde kinQhw"
      onClick={[Function]}
    >
      Load More
    </button>
  </div>
  <div
    className="sc-jcwofb cOCcZx"
  >
    <p>
      How are we doing? Give us feedback on this page.
    </p>
    <h2>
      Sign up for REI emails
    </h2>
    <p>
      Co-op offers, events & cool new gear
    </p>
    Email
    <form>
      <input
        type="text"
      />
      <button
        className="sc-hiKfjK eprGiy"
      >
        Sign me up!
      </button>
    </form>
  </div>
  <div
    className="sc-jJMGHv dyzYGl"
  >
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
    <p>
      Other Stuff
    </p>
  </div>
</div>
`;

exports[`renders Dropdown component correctly 1`] = `
<div>
  <div
    className="sc-iwaifL gUOPVN"
  >
    <p>
      Sort by: 
       ▾
    </p>
    <div
      className="sc-bCwgka kfElgi"
    >
      <div
        className="sc-cxNIbT ibZciV"
        onClick={[Function]}
      >
        <div>
          Newest questions
        </div>
      </div>
      <div
        className="sc-cxNIbT ibZciV"
        onClick={[Function]}
      >
        <div>
          Newest answers
        </div>
      </div>
      <div
        className="sc-cxNIbT ibZciV"
        onClick={[Function]}
      >
        <div>
          Most answered
        </div>
      </div>
      <div
        className="sc-cxNIbT ibZciV"
        onClick={[Function]}
      >
        <div>
          Answers needed
        </div>
      </div>
      <div
        className="sc-cxNIbT ibZciV"
        onClick={[Function]}
      >
        <div>
          Most helpful answers
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Helpful component correctly 1`] = `
<p>
  Helpful? 
  <button
    className="sc-bdnylx bGopLz yes"
    onClick={[Function]}
  >
    Yes · 
    0
  </button>
   
  <button
    className="sc-bdnylx bGopLz no"
    onClick={[Function]}
  >
    No · 
    1
  </button>
   
  <button
    className="sc-bdnylx bGopLz"
  >
    Report as inappropriate
  </button>
</p>
`;

exports[`renders QuestionList component correctly 1`] = `
<div>
  <section
    className="sc-fKgIGh keGHJm"
  >
    <p>
      1-
      2
       of 
      2
       Questions
    </p>
  </section>
  <div>
    <section
      className="sc-fFSRdu lTHh"
    >
      <div>
        <section
          className="sc-bkbjAj KBlYw"
        >
          1
          <div
            className="sc-kfYqjs jVfLBx"
          >
            answers
          </div>
        </section>
        <div>
          <div
            className="sc-hHEjAm hmcqje"
          >
            BobbyB
             · 
            <button
              className="sc-dIvqjp dOeeHJ"
            >
              17 minutes ago
            </button>
          </div>
        </div>
        <div
          className="sc-dlMBXb fghCqn"
        >
          <p>
            Does this product fit both men and women?
          </p>
        </div>
        <button
          className="sc-iemXMA kQNakb"
          onClick={[Function]}
        >
          Answer the question
        </button>
      </div>
      <div>
        <div>
          <div
            className="sc-jSFkmK crwJGT"
          >
            REI Helper Sally
             · 
            <button
              className="sc-eCApGN fooWmP"
            >
              17 minutes ago
            </button>
          </div>
        </div>
        <div
          className="sc-iCoHVE johfzo"
        >
          <p>
            Yes it does.
          </p>
        </div>
      </div>
      <p>
        Helpful? 
        <button
          className="sc-bdnylx bGopLz yes"
          onClick={[Function]}
        >
          Yes · 
          0
        </button>
         
        <button
          className="sc-bdnylx bGopLz no"
          onClick={[Function]}
        >
          No · 
          1
        </button>
         
        <button
          className="sc-bdnylx bGopLz"
        >
          Report as inappropriate
        </button>
      </p>
    </section>
  </div>
  <div>
    <section
      className="sc-fFSRdu lTHh"
    >
      <div>
        <section
          className="sc-bkbjAj KBlYw"
        >
          2
          <div
            className="sc-kfYqjs jVfLBx"
          >
            answers
          </div>
        </section>
        <div>
          <div
            className="sc-hHEjAm hmcqje"
          >
            Rachel098
             · 
            <button
              className="sc-dIvqjp dOeeHJ"
            >
              17 minutes ago
            </button>
          </div>
        </div>
        <div
          className="sc-dlMBXb fghCqn"
        >
          <p>
            How durable is this?
          </p>
        </div>
        <button
          className="sc-iemXMA kQNakb"
          onClick={[Function]}
        >
          Answer the question
        </button>
      </div>
      <div>
        <div>
          <div
            className="sc-jSFkmK crwJGT"
          >
            REI Helper Jim
             · 
            <button
              className="sc-eCApGN fooWmP"
            >
              17 minutes ago
            </button>
          </div>
        </div>
        <div
          className="sc-iCoHVE johfzo"
        >
          <p>
            It can stand up to all your favortie activities.
          </p>
        </div>
      </div>
      <p>
        Helpful? 
        <button
          className="sc-bdnylx bGopLz yes"
          onClick={[Function]}
        >
          Yes · 
          2
        </button>
         
        <button
          className="sc-bdnylx bGopLz no"
          onClick={[Function]}
        >
          No · 
          0
        </button>
         
        <button
          className="sc-bdnylx bGopLz"
        >
          Report as inappropriate
        </button>
      </p>
    </section>
  </div>
</div>
`;
